### ISSUE TITLE
Implement React 16 Error Boundaries

### ISSUE THREAD

**Issue by tisto** [plone/plone-react#57]
React 16 allows to Error Boundaries to catch errors in components:

https://reactjs.org/blog/2017/07/26/error-handling-in-react-16.html

This way the entire app will not break when there is an error in one component. Also, error reporting to Sentry becomes a lot easier.

**Comment by tulikavijay** [comment]
I would like to work on this. 

I am thinking of implementing this in the following way:

1.Creating a ErrorBoundary class component in the /src folder,
2.Using the error.jsx to display the error.
3.Using the ErrorBoundary in client.jsx ,in route.jsx for `<Route>` and in server.jsx here: 
```
const component = (
              <Provider store={store}>
                <RouterContext {...routeState} />
              </Provider>
);
```

**Comment by tisto** [comment]
@tulikavijay you can not use ErrorBoundaries in React 15 (which we use right now). We first have to upgrade to React 16 before we can proceed with this.

**Comment by tulikavijay** [comment]
thanks for the reply! i'll look at other issues.

**Comment by nileshgulia1** [comment]
I have a small approach from my side. We will wrap our different small components independently in different error boundaries so if one component crashes, the others will remain intact.
``` javascript 
<Errorbounday><child 1/></Errorboundary>
<Errorbounday><child 2/></Errorboundary>
```

**Comment by robgietema** [comment]
Fixed in #106
